samples/matter/thermostat/README.rst

```diff
@@ -0,0 +1,312 @@
+********************
+
+.. |sample path| replace:: :file:`samples/matter/thermostat`
+
+.. include:: /includes/build_and_run.txt
+
+Selecting a build type
+======================
+
+Before you start testing the application, you can select one of the `Matter thermostat build types`_, depending on your building method.
+Thermsotat sample has additional configuration which allows you to connect and test it with external matter sensor. For this option you need
+to select this option by adding the kconfig flag -DCONFIG_EXTERNAL_SENSOR=y.
+
+Selecting a build type in |VSC|
+-------------------------------
+
+.. include:: /config_and_build/modifying.rst
+   :start-after: build_types_selection_vsc_start
+   :end-before: build_types_selection_vsc_end
+
+Selecting a build type from command line
+----------------------------------------
+
+.. include:: /config_and_build/modifying.rst
+   :start-after: build_types_selection_cmd_start
+   :end-before: For example, you can replace the
+
+For example, you can replace the *selected_build_type* variable to build the ``release`` firmware for ``nrf52840dk_nrf52840`` by running the following command in the project directory:
+
+.. parsed-literal::
+   :class: highlight
+
+   west build -b nrf52840dk_nrf52840 -d build_nrf52840dk_nrf52840 -- -DCONF_FILE=prj_release.conf
+
+The ``build_nrf52840dk_nrf52840`` parameter specifies the output directory for the build files.
+
+.. note::
+   If the selected board does not support the selected build type, the build is interrupted.
+   For example, if the ``shell`` build type is not supported by the selected board, the following notification appears:
+
+   .. code-block:: console
+
+      File not found: ./ncs/nrf/samples/matter/thermostat/configuration/nrf52840dk_nrf52840/prj_shell.conf
+
+
+
+Testing
+=======
+
+When you have built the sample and programmed it to your development kit, it automatically starts the Bluetooth LE advertising and the **LED1** starts flashing (Short Flash On).
+Thermostat should update its values every 30s and print data to logs.
+By pressing **Button 2** you can print thermostat data to the logs.
+At this point, you can press **Button 1** for six seconds to initiate the factory reset of the device.
+
+
+.. _matter_thermostat_network_testing:
+
+Testing in a network
+--------------------
+
+To test the sample in a Matter-enabled Thread network, complete the following steps:
+
+.. matter_thermostat_sample_testing_start
+
+1. |connect_kit|
+#. |connect_terminal_ANSI|
+#. Press **Button 2** to print Thermostat data to logs.
+#. Press **Button 1** for six seconds to initiate the factory reset of the device.
+
+The device reboots after all its settings are erased.
+
+
+Testing with external sensor
+----------------------------
+
+   Thermostat sample can be connected with external sensor device (for example Thingy:53: Matter weather station). Connection is done by :ref:`matter_thermostat_sample_binding` to matter temperature measurement cluster.
```

Comment:
```suggestion
   Thermostat sample can be connected with external sensor device (for example Thingy:53: Matter weather station). Connection is done by :ref:`matter_thermostat_sample_binding` to Matter temperature measurement cluster.
```
Maybe we should link the:
[Thingy:53: Matter weather station](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/matter_weather_station/README.html)


Also the "_Connection is done by"_ sounds a bit weird, but maybe @greg-fer will comment on that.